src/core/document.js

```diff
@@ -422,58 +441,63 @@ class Page {
 
     // Fetch the page's annotations and add their operator lists to the
     // page's operator list to render them.
-    return Promise.all([pageListPromise, this._parsedAnnotations]).then(
-      function ([pageOpList, annotations]) {
+    return Promise.all([
+      pageListPromise,
+      this._parsedAnnotations,
+      newAnnotationsPromise,
+    ]).then(function ([pageOpList, annotations, newParsedAnnotations]) {
+      if (Array.isArray(newParsedAnnotations)) {
+        annotations = annotations.concat(newParsedAnnotations);
```

Comment:
Maybe slightly shorter names (again), and the `isArray`-check might be overkill (since it's either `null` or an Array):
```suggestion
    ]).then(function ([pageOpList, annotations, newAnnotations]) {
      if (newAnnotations) {
        annotations = annotations.concat(newAnnotations);
```